internal/uri/uri_util.go

```diff
@@ -48,15 +51,58 @@ func FolderContains(folderPath string, path string) bool {
 // PathFromUri converts the given uri to a file path
 func PathFromUri(documentURI sglsp.DocumentURI) string {
 	u := string(documentURI)
+
+	// Check if path is UNC file path. In this case return it.
+	uncPath := pathFromUNCUri(u)
+	if uncPath != "" {
+		return uncPath
+	}
+
 	if !strings.HasPrefix(u, fileScheme) && strings.HasPrefix(u, eclipseWorkspaceFolderScheme) {
 		u = strings.Replace(u, eclipseWorkspaceFolderScheme, fileScheme, 1)
 	}
+
 	return uri.New(u).Filename()
 }
 
+// isUNCFileURI checks if the provided file URI represents a UNC path.
+func pathFromUNCUri(uri string) string {
+	if runtime.GOOS != "windows" || !strings.HasPrefix(uri, fileScheme) {
+		return ""
+	}
+	if strings.HasPrefix(uri, UncFileScheme) {
+		uri = strings.Replace(uri, UncFileScheme, fileScheme, 1)
+	}
+
+	parsedURI, err := url.Parse(uri)
+	if err != nil {
+		return ""
+	}
+
+	if parsedURI.Scheme != "file" {
+		return ""
+	}
+
+	// A non-empty Host indicates a UNC path
+	if len(parsedURI.Host) > 0 {
+		uncPath := fmt.Sprintf(`\\%s%s`, parsedURI.Host, parsedURI.Path)
```

Comment:
Is `parsedURI.Path` guaranteed to be absolute? Will this mix forward and back-slashes?

Reply:
I now know why the Java people introduced the `Path` type. Mixing of slashes should be tackled in line 90. Maybe I'd do a `filepath.Clean` before returning.

Reply:
> // On Windows, Clean does not modify the volume name other than to replace
// occurrences of "/" with `\`.
// For example, Clean("//host/share/../x") returns `\\host\share\x`.

maybe we just need to use clean instead of normalizing the the slashes